packages/inngest/src/components/InngestCommHandler.ts

```diff
@@ -1234,10 +1235,27 @@ export class InngestCommHandler<
   }
 
   protected configs(url: URL): FunctionConfig[] {
-    return Object.values(this.rawFns).reduce<FunctionConfig[]>(
+    const configs = Object.values(this.rawFns).reduce<FunctionConfig[]>(
       (acc, fn) => [...acc, ...fn["getConfig"](url, this.id)],
       []
     );
+
+    for (const config of configs) {
+      try {
+        functionConfigSchema.parse(config);
+      } catch (e) {
+        if (e instanceof z.ZodError) {
+          this.log(
+            "error",
+            `Config invalid for function "${config.id}" : ${e.errors
+              .map((err) => err.message)
+              .join("; ")}`
+          );
+        }
+      }
+    }
```

Comment:
Nit: Could use `.safeParse()` to avoid having to use do the `instanceof ZodError` check:
```suggestion
    for (const config of configs) {
      const check = functionConfigSchema.safeParse(config);
      if (!check.success) {
        const errors = check.error.errors.map((err) => err.message).join("; ");

        this.log(
          "error",
          `Config invalid for function "${config.id}" : ${errors}`
        );
      }
    }
```